itests/eth_deploy_test.go

```diff
@@ -94,17 +96,25 @@ func TestDeployment(t *testing.T) {
 	hash := client.EVM().SubmitTransaction(ctx, &tx)
 	fmt.Println(hash)
 
+	mpoolTx, err := client.EthGetTransactionByHash(ctx, &hash)
+	require.NoError(t, err)
+
+	// require that the hashes are identical
+	require.Equal(t, hash, mpoolTx.Hash)
+
+	// these fields should be nil because the tx hasn't landed on chain.
```

Comment:
Yeah, let's not play this game, we're inviting flakiness here.

Can we just try to lookup the receipt before making these assertions? If found, let's simply skip them (with a log saying that we're doing so).

Reply:
This is actually a pretty important assertion (we had bugs here before, so want to lock these for regression testing). I'm onboard with deflaking, as long as we don't withdraw coverage we need.

Reply:
The Miners in the itest framework have `pause` and `unpause` functions exposed, but we should implement a function for the test kit that allows us to pause and unpause all mining.